### DISCUSSION TITLE
无法编译中文参考文献

### DISCUSSION THREAD

**Discussion by hzxscyq** [sjtug/SJTUThesis#1003] [提问]
windows上用vscode。之前bib文件都是英文文献，今天插入一个中文文献并在正文中用parencite引用后报错，错误只提示在main.tex的\printbibliography[heading=bibintoc]中提示

Argument of \__text_change_case_break:w has an extra }.
<inserted text> 
LaTeX
Paragraph ended before \__text_change_case_break:w was complete.
<to be read again> 
LaTeX
Missing number, treated as zero.
<to be read again> 

出现这个错误后，我把bib条目里所有中文内容替换成英文’test‘也依然会有同样错误。我必须在正文里删掉引用重新编译才能正常工作。

**Comment by hushidong** [comment]
把导致错误的中文文献的bib发出来看一下

获取Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
From: hzxscyq ***@***.***>
Sent: Tuesday, January 9, 2024 8:58:27 PM
To: sjtug/SJTUThesis ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [sjtug/SJTUThesis] 无法编译中文参考文献 (Discussion #1003)


windows上用vscode。之前bib文件都是英文文献，今天插入一个中文文献并在正文中用parencite引用后报错，错误只提示在main.tex的\printbibliography[heading=bibintoc]中提示

Argument of __text_change_case_break:w has an extra }.

LaTeX
Paragraph ended before __text_change_case_break:w was complete.

LaTeX
Missing number, treated as zero.

出现这个错误后，我把bib条目里所有中文内容替换成英文’test‘也依然会有同样错误。我必须在正文里删掉引用重新编译才能正常工作。

―

**Reply by AlexaraWu** [reply]
我在 TeX Live 2021 上使用 `pdflatex` 编译中文文档的时候可以复现这个问题。

```latex
% !TeX encoding = UTF-8

\documentclass[UTF8,fontset=windows]{ctexart}

\begin{filecontents}[force]{\jobname.bib}
@article{Li1999,
  author  = {李晓东 and 张庆红 and 叶瑾琳},
  title   = {气候学研究的若干理论问题},
  journal = {北京大学学报: 自然科学版},
  year    = {1999},
  volume  = {35},
  number  = {1},
  pages   = {101--106}
}

\end{filecontents}

\usepackage[backend=biber,style=gb7714-2015]{biblatex}
\addbibresource{\jobname.bib}

\begin{document}

文献\cite{Li1999}。

\printbibliography

\end{document}
```

```console
(./main.bbl) (/usr/local/texlive/2021/texmf-dist/tex/latex/ctex/fd/c70rm.fd)
! Argument of \__text_change_case_break:w has an extra }.
<inserted text> 
                \par 
l.26 
     
? 
```

使用 `xelatex` 或 `lualatex` 编译时不会有这个问题。

<table>
  <thead>
    <tr>
      <td colspan="2">TeX Live 2021</td>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><code>ctex</code></td>
      <td>v2.5.6</td>
    </tr>
    <tr>
      <td><code>biblatex</code></td>
      <td>v3.16</td>
    </tr>
    <tr>
      <td><code>biblatex-gb7714-2015</code></td>
      <td>v1.0w</td>
    </tr>
  </tbody>
</table>

**Reply by hzxscyq** [reply]
@artical{zhu2022wiretap,
  title={测试},
  author={测试},
  journal={test},
  volume={33},
  number={5},
  pages={710--711},
  year={2022}
}

这个本地Latexmk编译就会报错。但我上传到overleaf后编译就没问题了

**Reply by hushidong** [reply]
texlive2021，已经没有环境测试了。
我试了下texlive2023是没有问题的。

那么你试一下更新宏包试试？先更新biblatex-gb7714-2015，若不行再更新一下biblatex，若实在不行，还是安装一下texlive2023吧，让电脑安装自己跑1个小时，安装就好了。

**Reply by AlexaraWu** [reply]
> texlive2021，已经没有环境测试了。 我试了下texlive2023是没有问题的。

我本地装了多个版本，做模板兼容性测试用的。之前测 TeX Live 2021 的时候碰到过这个问题，顺便在这里提一下。

我估计题主遇到这个问题可能也是版本问题，不妨说明一下本地的编译环境及编译方式。

**Reply by hushidong** [reply]
嗯，题主可以拷贝最新的gb7714-2015.cbx/bbx到你的工作目录下，试试看，是否会解决。这样也等于更新了biblatex-gb7714-2015
